test/e2e/checkpoint_test.go

```diff
@@ -41,11 +42,12 @@ var _ = Describe("Podman checkpoint", func() {
 		if !criu.CheckForCriu() {
 			Skip("CRIU is missing or too old.")
 		}
-		// TODO: Remove the skip when the current CRIU SELinux problem is solved.
-		// See: https://github.com/containers/libpod/issues/2334
+		// Only Fedora 29 and newer has a new enough selinux-policy and
+		// container-selinux package to support CRIU in correctly
+		// restoring threaded processes
 		hostInfo := podmanTest.Host
-		if hostInfo.Distribution == "fedora" {
-			Skip("Checkpointing containers on Fedora currently broken.")
+		if hostInfo.Distribution == "fedora" && hostInfo.Version < "29" {
+			Skip("Checkpoint/Restore with SELinux only works on Fedora >= 29")
```

Comment:
What about Ubuntu?

Consider flipping the distribution logic.

```golang
if hostInfo.Distribution != "fedora" or hostInfo.Version < "29" {
    skip(...)
}
```

Reply:
errr...does this work on Ubuntu?

Reply:
This works on Ubuntu. The main blocker are SELinux related changes in CRIU, container-selinux, selinux-policy.